### ISSUE TITLE
Next / Previous doesn't work when looking at a message inside a bundle

### ISSUE THREAD

**Issue by leggett** [teresa-ou/inboxy#48]
Really impressive extension you've built. One thing that doesn't seem to work is the re-ordering of the threads with flex order doesn't change the order of the messages when going next/previous inside a message. 

**Steps to reproduce:**

1. Expand bundle
2. Click on the first message
3. Click the next message button or hit [J]
4. Get a message different than the one that was next in the bundle. --- Expected behavior: Get the next message in the bundle

**Possible incomplete solution:** 
One way I can think of fixing this is basically intercepting and replacing the next/previous action (either click on button or hitting J/K). The list is still loaded in the DOM and I've confirmed you can simulate a click on a thread in the list as a way of jumping to a specific message. 

Some things to note/consider:

- What about the setting to go to the next message upon archiving or deleting a message? Or that combined with the "Send + Archive" button. 

- Important to note that the Next/Prev buttons aren't always between messages. In Reading Pane, they navigate between the different pages of the list and there is no Next/Prev message button.

- This breaks if you're at the end of the list and the action is meant to go to the 101st tread or if you're on page 2 (threads 101-200) and want to go to the 100th tread. Maybe you conditionally block the action and let Gmail handle Next/Prev to cover this case?

**Comment by teresa-ou** [comment]
Thanks for pointing this out and for the detailed writeup! 

I think that while the solution you mentioned could work, it also makes the extension more complicated. I'd be curious to see if many people use the next/prev buttons enough to justify adding this. 

Personally, I open an email, then use the back button on my mouse to go back to the bundle to open another email. But that does feel unwieldy and takes 2 clicks instead of 1.

**Comment by avamsi** [comment]
+1, I use j/k and {/} all the time and was questioning if something was wrong with my filters -- I then realized those actions were WAI, just not within the bundle but based on the pre-bundled order.

**Comment by sttts** [comment]
This also breaks the archive function when it is set up to jump to the next message (I think that can be configured in gmail). Now it jump to the next message chronologically, not the next in the bundle. So I think it's the same root cause as in this issue.

**Comment by kahliloppenheimer** [comment]
+1 I'm a big user of j/k for quickly going through and archiving emails. Would love to see that awesome addition to an already amazing extension :).

**Comment by R-XS** [comment]
I agree with the others, the current behaviour of previous and next, as well as archive+next are confusing.

**Comment by samuelalexmclean** [comment]
Another +1. Amazing extension, but this would make it even... amazinger.

**Comment by butterflyhug** [comment]
FYI, up/down keyboard focus navigation through the message list itself is similarly causing the focus to bounce around because it’s visiting messages in their original order.

**Comment by CaneRaiser** [comment]
One possible workaround would be to have an option to go straight to the "view all" screen when a bundle is clicked rather than expanding the bundle inline. Since that screen is a standard Gmail view of all of the messages that are both in the inbox and have that label, all of the keyboard shortcuts work just fine there.   Bonus points if there's a way to activate the bundle from the message list while browsing the message list with keyboard commands....

**Comment by leggett** [comment]
I'm finally working on adding bundling into [Simplify Gmail](https://simpl.fyi) and I've figured out how to support `j` and `k` (as well as the up/down arrows). 

Digging deep into this issue, I tried everything I could think of to avoid my original suggestion. But alas, the only solution that worked was the hard one: intercept the next/previous keyboard shortcuts and entirely replace them so the correct message is focused (in the list view) or opened (in message view). It was a beast to get working but is possible.

**Comment by benoror** [comment]
Hi! Keyboard next/previous (`j` / `k` and the UI buttons) not following visual bundle order is a known hard problem (DOM order vs flex order). It is **not** fixed in **Inbundly** yet.

I would still love a fresh issue on the Inbundly tracker if this blocks you, especially with "reading pane on/off" notes. Related shipping features: **remember open bundle**, **select-all**, **snooze whole bundle**.

Chrome: https://chromewebstore.google.com/detail/inbundly/pbfjicjmcpogjlbpljebhhgkgfhbdcga
Issues: https://github.com/benoror/inbundly/issues/new
